modules/@apostrophecms/piece-type/index.js

```diff
@@ -198,7 +198,24 @@ module.exports = {
     },
     async getOne(req, _id) {
       self.publicApiCheck(req);
-      const doc = await self.getRestQuery(req).and({ _id }).toObject();
+
+      const commonValues = {
+        any: null,
+        false: false,
+        true: true
+      };
+      const publishedValues = {
+        null: true,
+        ...commonValues
+      };
+      const trashValues = {
+        null: false,
+        ...commonValues
+      };
+
+      const published = publishedValues[req.query.published];
+      const trash = trashValues[req.query.trash];
+      const doc = await self.getRestQuery(req).and({ _id }).published(published).trash(trash).toObject();
```

Comment:
You don't need all this req.query stuff, getRestQuery already does it:

`query.applyBuildersSafely(req.query);`

Reply:
I like what you did here in general though. I hesitated to suggest it before, but this does feel a little more intuitive to me than the alternative: the default rules for visibility of a piece shouldn't be different for getAll and getOne.